diff --git a/.config/xkb/xkbdump b/.config/xkb/xkbdump
index 4196516..05365fd 100644
--- a/.config/xkb/xkbdump
+++ b/.config/xkb/xkbdump
@@ -1217,7 +1217,7 @@ xkb_symbols "pc+us+ru:2+sd:3+inet(evdev)+group(menu_switch)+capslock(grouplock)"
     key <AE11> {         [           minus,      underscore ] };
     key <AE12> {         [           equal,            plus ] };
     key <BKSP> {         [       BackSpace,       BackSpace ] };
-    key  <TAB> {         [             Tab,    ISO_Left_Tab ] };
+    key  <TAB> {         [ ISO_First_Group,  ISO_Last_Group ] };
     key <AD01> {
         type= "ALPHABETIC",
         symbols[Group1]= [               q,               Q ],
@@ -1291,7 +1291,11 @@ xkb_symbols "pc+us+ru:2+sd:3+inet(evdev)+group(menu_switch)+capslock(grouplock)"
         symbols[Group3]= [    bracketright,      braceright ]
     };
     key <RTRN> {         [          Return ] };
-    key <LCTL> {         [       Control_L ] };
+    key <LCTL> {
+        type= "ONE_LEVEL",
+        symbols[Group1]= [           ISO_Group_Latch ],
+        symbols[Group2]= [           Mode_switch ]
+    };
     key <AC01> {
         type= "ALPHABETIC",
         symbols[Group1]= [               a,               A ],
@@ -1440,7 +1444,11 @@ xkb_symbols "pc+us+ru:2+sd:3+inet(evdev)+group(menu_switch)+capslock(grouplock)"
     };
     key <LALT> {         [           Alt_L,          Meta_L ] };
     key <SPCE> {         [           space ] };
-    key <CAPS> {         [ ISO_First_Group,  ISO_Last_Group ] };
+    replace key <CAPS> {
+        type[Group1] = "ONE_LEVEL",
+        symbols[Group1] = [ Control_L ],
+        actions[Group1] = [ SetMods(modifiers=Control) ]
+    };
     key <FK01> {
         type= "CTRL+ALT",
         symbols[Group1]= [              F1,              F1,              F1,              F1, XF86Switch_VT_1 ]
